pandas/core/generic.py

```diff
@@ -4533,6 +4533,8 @@ def sort_index(
 
         inplace = validate_bool_kwarg(inplace, "inplace")
         axis = self._get_axis_number(axis)
+        ascending = validate_ascending(ascending)
```

Comment:
you don't need allow of this, just call `validate_bool_kwarg`

Reply:
``validate_bool_kwargs`` accepts ``None`` as a valid value.
Do we consider ``ascending=None`` as valid?
https://github.com/pandas-dev/pandas/issues/39434#issuecomment-768407714

Reply:
in this case no
so i can enhance the checker with an arg to only allow an actual bool

Reply:
pls use `validate_bool_kwarg` instead of adding *anything* here. you can add a kwargs to that function itself to make things strict

Reply:
``ascending`` here may be a single value or a sequence of items.
I am afraid that I still need to add validation of either a scalar value or each of the items in the sequence.

Reply:
in the def we have `ascending: bool_t = True`, so should probably update that too as part of this PR.

Reply:
Extended typing for ``ascending``